renderers/css/actions.py

```diff
@@ -36,13 +36,16 @@ class OpiExecuteCommand(OpiAction):
 class OpiOpen(OpiAction):
     """Renderer for write PV actions."""
     ACTION_TYPE = 'OPEN_DISPLAY'
+    MACRO_ERROR = 'Macros must have key and value as strings ({}:{} violates this rule)'
 
     def render(self, actions_node, action_model):
         action_node = super(OpiOpen, self).render(actions_node, action_model)
         macros_node = et.SubElement(action_node, 'macros')
         parent_macros_node = et.SubElement(macros_node, 'include_parent_macros')
         parent_macros_node.text = 'true' if action_model._parent_macros else 'false'
         for key, value in action_model._macros.items():
+            if not utils.is_string(key) or not utils.is_string(value):
```

Comment:
If a user passes anything that isn't a string as a macro `value` they would accept it being naively rendered as `__str__` (a note in the doc strings would make that explicit).
This limits the checking you need to do to ensuring the macro name (`key`) is valid XML tag.